topup/handler_test.go

```diff
@@ -0,0 +1,145 @@
+package topup_test
+
+import (
+	"math/big"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ethTypes "github.com/maticnetwork/bor/core/types"
+	"github.com/maticnetwork/heimdall/contracts/stakinginfo"
+	"github.com/maticnetwork/heimdall/topup/types"
+	hmTypes "github.com/maticnetwork/heimdall/types"
+	"github.com/tendermint/tendermint/crypto/secp256k1"
+
+	"github.com/maticnetwork/heimdall/app"
+	"github.com/maticnetwork/heimdall/helper/mocks"
+	"github.com/maticnetwork/heimdall/topup"
+	"github.com/maticnetwork/heimdall/types/simulation"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+// HandlerTestSuite integrate test suite context object
+type HandlerTestSuite struct {
+	suite.Suite
+
+	app            *app.HeimdallApp
+	ctx            sdk.Context
+	cliCtx         context.CLIContext
+	querier        sdk.Querier
+	handler        sdk.Handler
+	contractCaller mocks.IContractCaller
+}
+
+// SetupTest setup all necessary things for querier tesing
+func (suite *HandlerTestSuite) SetupTest() {
+	suite.app, suite.ctx, suite.cliCtx = createTestApp(false)
+	suite.contractCaller = mocks.IContractCaller{}
+	suite.querier = topup.NewQuerier(suite.app.TopupKeeper, &suite.contractCaller)
```

Comment:
No need to create querier to test handler.

Reply:
my bad 👍